src/libstd/fs.rs

```diff
@@ -1069,13 +1069,23 @@ pub fn remove_dir_all<P: AsRef<Path>>(path: P) -> io::Result<()> {
 }
 
 fn _remove_dir_all(path: &Path) -> io::Result<()> {
+    let filetype = try!(symlink_metadata(path)).file_type();
+    if filetype.is_symlink() {
+        remove_file(path)
```

Comment:
I've confirmed that directory symlinks on windows cannot be removed by `remove_file` but can be removed via `remove_dir`.